PowerPointLabs/PowerPointLabs/Models/PowerPointShapeGalleryPresentation.cs

```diff
@@ -562,13 +563,27 @@ private bool InitCategories()
             bool shapeLost = false;
             bool pngLost = false;
             int untitledCategoryCnt = 0;
+            List<int> inaccessibleCategoryIndices = new List<int>();
+            List<PowerPointSlide> slides = Slides;
 
-            foreach (PowerPointSlide category in Slides)
+            for (int i = 0; i < Slides.Count; i++)

```

Comment:
You can iterate backwards instead of forwards and I think that can save the need for the `List<int> inaccessibleCategoryIndices`

By iterating backwards, you can safely delete the slide